### PR TITLE
add dockerfile

### PR DESCRIPTION
**Pull request by abbbi**
in the next step we could automatically release docker image on ghcr.io during release using github workflow.

### CODE REVIEW

**Comment by abbbi** [comment]
some suggestions applied.

**Comment by mmomjian** [comment]
May want to add `EXPOSE 8007`, while this is not needed for functionality it helps to document the port used by the container.

File: Dockerfile [hunk 1]
```diff
@@ -8,25 +8,22 @@
 #  docker run -it  pmoxs3backuproxy:latest pmoxs3backuproxy -endpoint 127.0.0.1:9000
 #
 # Note: uses default server key and certificate stored in /root/
-FROM debian:bookworm-slim
+FROM golang:1.22.5 as build
```
Comment (mmomjian) [inline Dockerfile:11]: should be `FROM .... AS build` (note uppercase)

Comment (abbbi) [reply Dockerfile:11]: works for me with Docker version 20.10.24+dfsg1, build 297e128 nevertheless.

Comment (mmomjian) [reply Dockerfile]: I think it still works but it produces a build-time error

Comment (abbbi) [reply Dockerfile]: pushed 024ca29

**Comment by mmomjian** [comment]
I have a working Dockerfile for local building if anyone would like it:

Add the following to the top of the file:

```
FROM alpine/git AS source
RUN cd / && git clone 'https://github.com/tizbac/pmoxs3backuproxy'
```

Then add the following under the LABELs instead of the old `COPY`

`COPY --from=source /pmoxs3backuproxy .`

**Comment by abbbi** [comment]
#13

**Pull request closed without merging**